src/features/ci-cd/hooks/service-installs.ts

```diff
@@ -65,35 +54,54 @@ const createReleaseServiceInstalls = async (
 	}
 	const serviceIds = services.map(({ id }) => id);
 
-	const serviceInstalls = (await api.get({
-		resource: 'service_install',
+	const missingServiceFilters = serviceIds.map((serviceId) => ({
+		$not: {
+			service_install: {
+				$any: {
+					$alias: 'si',
+					$expr: {
+						si: {
+							installs__service: serviceId,
+						},
+					},
+				},
+			},
+		},
+	}));
+
+	const devicesToAddServiceInstalls = (await api.get({
```

Comment:
For the case that a fleet changes its target release while the two releases share the same services, this previously was returning `D * S` rows, while now it would return 0.
As a result that also reduces the `O(D * S)` complexity from the `.map`s & `_.difference` of the code running in the `Promise.all` (which in the above scenario would have resulted 0 INSERTs, since everything would match).